### PR TITLE
feat(developer): kmcomp x64

### LINKED ISSUE #2350: [Developer] kmcomp x64
**Is your feature request related to a problem? Please describe.**
On macOS 10.15 and newer Linux distributions, x86 executables are no longer supported. This means that WINE wrapper for kmcomp.exe no longer works.

This is a blocking issue for FV Keyboards app because it depends on having the keyboard compiler available via WINE in order to build the app for iOS.

**Describe the solution you'd like**
We need an x64 version of kmcomp.exe and kmcmpdll.dll. Even better, native versions of the programs (although this would be much more work, of course).

**Describe alternatives you've considered**
In theory, the FV Keyboards app could pull precompiled versions of the keyboards from downloads.keyman.com/keyboards but given this is a breaking issue for all other macOS users anyway, that's only a stopgap.

**Product context**
 - OS: macOS 10.15

### PR DESCRIPTION
**Pull request by mcdurdin**
Fixes #2350.

This PR adds support for x64 in kmcomp and kmcmpdll.

See also keymanapp/keyboards#984, keymanapp/keyboards-starter#11

A full regression test has been implemented in the keyboards PR.

---

Completed: 
- [x] Fixup structure sizes in kmcmpdll
- [x] Audit kmcomp for structure size changes (e.g. kmx load for kmw compile)
- [x] Create e2e or unit tests for file structure writes
- [x] Run a test against keyboards repo to validate 100% identical outputs between 32 and 64 bit
- [x] Validate location of files and inclusion in installers and distribution
- [x] Update keyboards repo to use either x64 or x86 version of compiler depending on context
- [x] Assess difficulty in building kmcomp + kmcmpdll as cross-platform, so as to avoid WINE dependency

### CODE REVIEW

**Comment by mcdurdin** [comment]
> Assess difficulty in building kmcomp + kmcmpdll as cross-platform, so as to avoid WINE dependency

So, Delphi requires both a Windows + mac, and Windows + Linux in order to build cross platform executables. This adds a logistical nightmare to the CI that I just don't want to consider at this time!

**Comment by darcywong00** [comment]
I get the same usage with `kmcomp` and `kmcomp.x64`.
```
Keyman Developer Compiler
Version 13.0.9999.0, Copyright c SIL International.  All Rights Reserved.

Usage: kmcomp [-s[s]] [-nologo] [-c] [-d] [-w] [-cfc] [-v[s|d]] [-source-path path] [-schema-path path]
              [-m] infile [-m infile] [-t target] [outfile.kmx|outfile.js [error.log]]
              [-add-help-link path] [-color|-no-color]
              [-extract-keyboard-info field[,field...]]
```

1. Should kmcomp.x64 usage have `Usage: kmcomp.x64`?
2. Would it be clearer for `kmcomp.x64` to have a description `Keyman Developer Compiler (64-bit)`.

**Comment by darcywong00** [comment]
Is Developer going to start using kmcomp.x64 instead of kmcomp? (Maybe a separate PR?)

**Comment by mcdurdin** [comment]
> Is Developer going to start using kmcomp.x64 instead of kmcomp? (Maybe a separate PR?)

No. Developer uses kmcmpdll.dll (32-bit) internally and won't be changing: the only reason we are creating an x64 version is system compatibility, not because we are coming anywhere near running out of memory in the 32 bit version (we probably use <100MB).

File: windows/src/Defines.mak [hunk 1]
```diff
@@ -146,6 +146,7 @@ TARGET_PATH=Release
 
 WIN32_TARGET_PATH=Win32\$(TARGET_PATH)
 WIN64_TARGET_PATH=Win64\$(TARGET_PATH)
+X64_TARGET_PATH=x64\$(TARGET_PATH)
```
Comment (darcywong00) [inline windows/src/Defines.mak]: I take it WIN64_TARGET is for something different from X64_TARGET?

Comment (mcdurdin) [reply windows/src/Defines.mak]: It's a kinda annoying situation where Visual C++ uses names the platform "x64" whereas Delphi names it "win64". I haven't found a clean way to make them mesh better.

File: windows/src/developer/kmcomp/Makefile [hunk 2]
```diff
@@ -29,5 +44,6 @@ test-manifest:
```
Comment (darcywong00) [inline windows/src/developer/kmcomp/Makefile]: Checking that you don't need `backup:` or `test-manifest:` for kmcomp.x64.exe?

**Pull request merged**